tests/wl_output.cpp

```diff
@@ -39,3 +39,15 @@ TEST_F(WlOutputTest, wl_output_properties_set)
     EXPECT_THAT((bool)output.mode_size, Eq(true));
     EXPECT_THAT((bool)output.scale, Eq(true));
 }
+
+TEST_F(WlOutputTest, wl_output_release)
+{
+    wlcs::Client client{the_server()};
+
+    // Skip test if compositor doesn't support a new enough wl_output version
+    client.acquire_interface(wl_output_interface.name, &wl_output_interface, WL_OUTPUT_RELEASE_SINCE_VERSION);
+
+    ASSERT_THAT(client.output_count(), Ge(1u));
+    client.release_output(0);
+    client.roundtrip();
+}
```

Comment:
I think this would be clearer as:
```
// Acquire *any* wl_output; we don't care which
auto output = static_cast<wl_output*>(client.acquire_interface(wl_output_interface.name, &wl_output_interface, WL_OUTPUT_RELEASE_SINCE_VERSION));
wl_output_release(output);
client.roundtrip()
```

You could then drop the `client.release_output()` implementation entirely.

Reply:
That's what I tried first. The problem is that since `wl_output.release` is marked as a destructor in the protocol, it is invalid to `wl_output_destroy()` it after a release (it segfaults). Thus the `client.release` which removes the output from the list and prevents it from being destroyed later.

Reply:
But `client.acquire_interface(...)` returns a *new* `wl_resource` that won't be freed by any other code; nothing will attempt to call `wl_output_destroy()` on it.

Reply:
Good point. Done